src/Aspire.Hosting.AWS/Utils/ProjectUtilities.cs

```diff
@@ -0,0 +1,123 @@
+        root["profiles"] = profiles;  // Ensure it's added to the root
+
+        var launchSettingsNodeKey = $"{Constants.LaunchSettingsNodePrefix}{resourceName}";
+
+        // Get or create the specific profile for Amazon.Lambda.TestTool
+        JsonObject? lambdaTester = profiles[launchSettingsNodeKey]?.AsObject();
+        if (lambdaTester == null)
+        {
+            lambdaTester = new JsonObject
+            {
+                ["commandName"] = "Executable",
+                ["executablePath"] = "dotnet"
+            };
+
+            profiles[launchSettingsNodeKey] = lambdaTester;
+        }
+
+        // Update properties that contain a path that is environment-specific
+        lambdaTester["commandLineArgs"] =
+            $"exec --depsfile ./{assemblyName}.deps.json --runtimeconfig ./{assemblyName}.runtimeconfig.json {SubstituteHomePath(runtimeSupportAssemblyPath)} {functionHandler}";
+        lambdaTester["workingDirectory"] = Path.Combine(".", "bin", "$(Configuration)", targetFramework);
+
+        // Serialize the updated JSON with indentation
+        var options = new JsonSerializerOptions { WriteIndented = true };
+        string updatedJson = root.ToJsonString(options);
+
+        // Save the updated JSON back to the launch settings file.
+        SaveLaunchSettings(projectPath, updatedJson);
+    }
+    
+    /// <summary>
+    /// Check if a path is in the user profile directory and use the environment-specific environment variable.
+    /// </summary>
+    /// <param name="path">The path to update to use the user profile environment variable</param>
+    /// <returns>A path that uses the user profile environment variable</returns>
+    private static string SubstituteHomePath(string path)
+    {
+        var userProfileEnvironmentVariable = "%USERPROFILE%";
+        var userProfilePath = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            userProfileEnvironmentVariable = "$HOME";
+        }
+
+        if (path.StartsWith(userProfilePath))
+        {
+            return path.Replace(userProfilePath, userProfileEnvironmentVariable);
+        }
+
+        return path;
+    }
+    
+    /// <summary>
+    /// Retrieve a project's launchSettings.json file contents
+    /// </summary>
+    /// <param name="projectPath">The project file path</param>
+    /// <returns>The launchSetting.json content</returns>
+    private static string GetLaunchSettings(string projectPath)
+    {
+        var parentDirectory = Path.GetDirectoryName(projectPath);
+        if (string.IsNullOrEmpty(parentDirectory))
+            throw new ArgumentException($"The project path '{projectPath}' is invalid. Unable to retrieve the '{Constants.LaunchSettingsFile}' file.");
+        var properties = Path.Combine(parentDirectory, "Properties");
+        if (!Directory.Exists(properties))
+        {
+            Directory.CreateDirectory(properties);
```

Comment:
so i understand in this case the launch settings doesnt exist. im not sure if we should be creating a directory though here. i assume you do this because when you end up writing the new launch settings you need the directory to exist right? maybe have the creation of the directory in there.

Reply:
I agree that it was surprising the `GetLaunchSettings` creates the directory. I think you can move that to the `SaveLaunchSettings` which is where I would expect the creation to exist.